sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -2345,20 +2344,15 @@ pi_result piMemBufferCreate(pi_context Context, pi_mem_flags Flags, size_t Size,
     //
   }
 
+  pi_result Result;
   if (DeviceIsIntegrated) {
-    ze_host_mem_alloc_desc_t ZeDesc = {};
-    ZeDesc.flags = 0;
-
-    ZE_CALL(zeMemAllocHost, (Context->ZeContext, &ZeDesc, Size, 1, &Ptr));
-
+    Result = piextUSMHostAlloc(&Ptr, Context, nullptr, Size, 64);
   } else {
-    ze_device_mem_alloc_desc_t ZeDesc = {};
-    ZeDesc.flags = 0;
-    ZeDesc.ordinal = 0;
-
-    ZE_CALL(zeMemAllocDevice,
-            (Context->ZeContext, &ZeDesc, Size, 1, ZeDevice, &Ptr));
+    Result = piextUSMDeviceAlloc(&Ptr, Context, Context->Devices[0], nullptr,
+                                 Size, 64);
```

Comment:
```suggestion
                                 Size, Size);
```
Wouldn't natural alignment be enough?

Reply:
I really meant to suggest min(Size,64) so if data fits a cache line it never crosses it, it is also less waste for Size<64